pkgs/by-name/vr/vrcadvert/deps.nix

```diff
@@ -8,11 +8,81 @@
     version = "1.0.70";
     hash = "sha256-Hl6ZmKBbS6YZX7cc1Jp4/Hz6ksZqlFR7ZllbZgHzeYw=";
   })
+  (fetchNuGet {
+    pname = "Microsoft.AspNetCore.App.Ref";
+    version = "6.0.36";
+    hash = "sha256-9jDkWbjw/nd8yqdzVTagCuqr6owJ/DUMi4BlUZT4hWU=";
+  })
+  (fetchNuGet {
+    pname = "Microsoft.AspNetCore.App.Runtime.linux-arm64";
+    version = "6.0.36";
+    hash = "sha256-JQULJyF0ivLoUU1JaFfK/HHg+/qzpN7V2RR2Cc+WlQ4=";
+  })
+  (fetchNuGet {
+    pname = "Microsoft.AspNetCore.App.Runtime.linux-x64";
```

Comment:
Running fetch-deps again got rid of those, thanks!

Reply:
> It looks like got rid of AspNetCore, but not NETCore?
> 
> I wonder if you need something like `-p:RuntimeFrameworkVersion=${dotnet-runtime.version}`?

You're right, there's still .NETCore 6 deps. Using that `dotnetFlags` makes `fetch-deps` fail:

```
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.osx-x64 with version (= 8.0.404) [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102:   - Found 177 version(s) in nuget [ Nearest version: 9.0.0-preview.1.24080.9 ] [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102:   - Found 1 version(s) in _nix [ Nearest version: 8.0.11 ] [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102: Unable to find package Microsoft.AspNetCore.App.Runtime.osx-x64 with version (= 8.0.404) [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102:   - Found 177 version(s) in nuget [ Nearest version: 9.0.0-preview.1.24081.5 ] [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102:   - Found 1 version(s) in _nix [ Nearest version: 8.0.11 ] [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Host.linux-x64 with version (= 8.0.404) [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102:   - Found 179 version(s) in nuget [ Nearest version: 9.0.0-preview.1.24080.9 ] [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102:   - Found 1 version(s) in _nix [ Nearest version: 8.0.11 ] [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102: Unable to find package Microsoft.NETCore.App.Host.osx-x64 with version (= 8.0.404) [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102:   - Found 179 version(s) in nuget [ Nearest version: 9.0.0-preview.1.24080.9 ] [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.csproj : error NU1102:   - Found 1 version(s) in _nix [ Nearest version: 8.0.11 ] [/tmp/fetch-deps-vrcadvert-1.0.0.4uT76P/source/VrcAdvert.sln]
```

But with the version from the error message it works, I'm not sure where are these versions are coming from, but pinning to an exact minor version doesn't seem that good of an idea: `dotnetFlags = [ "-p:RuntimeFrameworkVersion=8.0.11" ];`

Should we leave the .NETCore 6 deps in as before, pin to a specific minor version or do you have another idea?

Reply:
Pinning it is fine, but you'll want to pin it to the actual runtime being passed to `dotnet-runtime`.  If you're passing an SDK there it's not going to work because the sdk versions are different.   In that case you can use `dotnet-sdk.runtime.version`.

```
nix-repl> dotnet-sdk_8.version
"8.0.404"
nix-repl> dotnet-runtime_8.version
"8.0.11"
nix-repl> dotnet-sdk_8.runtime.version
"8.0.11"
```